protostar/composition_root.py

```diff
@@ -107,7 +107,7 @@ def build_di_container(
         configuration_file=configuration_file,
     )
 
-    project_compiler = ProjectCompiler(
+    contract_path_resolver = ContractPathResolver(
```

Comment:
I don't see a reason to pass this object around instead of a simple map of values, it would make migrating this to rust easier down the line.

Reply:
The problem is that what this module does is actually load contracts from ConfigurationFile + some extra error handling. We could replace the `ConfigurationFile` abstraction with a simple map, but I'm not sure if that would be a simple change, especially since we depend on updating this "file" in tests.

Reply:
If it's worth refactoring at all, we should consider changing tests as well then. Repacking it into object with another name is not really a huge improvement to the structure if i'm honest, but packing things into a map would actually simplify reasoning about the code a lot. The error handling can (and should be) part of validation/parsing of the configuration file.

Reply:
We've agreed to merge this as is and continue with refactoring when resolving #1882